components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/AdminAdvisoryDataRetrievalClient.java

```diff
@@ -0,0 +1,134 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.mgt.endpoint.util.client;
+
+import org.apache.axis2.transport.http.HTTPConstants;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.wso2.carbon.identity.mgt.endpoint.util.IdentityManagementEndpointUtil;
+import org.wso2.carbon.identity.mgt.endpoint.util.IdentityManagementServiceUtil;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+
+/**
+ * Client to interact with the Admin Advisory Management API.
+ */
+public class AdminAdvisoryDataRetrievalClient {
+
+    private static final String CLIENT = "Client ";
+    private static final Log LOG = LogFactory.getLog(AdminAdvisoryDataRetrievalClient.class);
+    private static final String ADMIN_BANNER_API_RELATIVE_PATH = "/api/server/v1/admin-advisory-management/banner";
+
+    /**
+     * Check for admin advisory banner configs in the given tenant.
+     *
+     * @param tenant Tenant Domain.
+     * @return A JSON Object containing admin advisory banner configs.
+     * @throws PreferenceRetrievalClientException Error while retrieving the admin advisory banner configs.
+     */
+    public JSONObject getAdminAdvisoryBannerData(String tenant)
+            throws PreferenceRetrievalClientException {
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
+
+            String uri = getAdminAdvisoryBannerEndpoint(tenant);
+
+            HttpGet request = new HttpGet(uri);
+            setAuthorizationHeader(request);
+
+            JSONObject jsonResponse = new JSONObject();
+
+            try (CloseableHttpResponse response = httpclient.execute(request)) {
+
+                if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                    jsonResponse = new JSONObject(
+                            new JSONTokener(new InputStreamReader(response.getEntity().getContent())));
+                }
+
+                return jsonResponse;
+            } finally {
+                request.releaseConnection();
+            }
+        } catch (IOException e) {
+            String msg = "Error while getting admin advisory banner preference for tenant : " + tenant;
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(msg, e);
+            }
+
+            throw new PreferenceRetrievalClientException(msg, e);
```

Comment:
Let's avoid logging and throwing since it is an anti-pattern. we should log and re-throw only if we are adding additional information to the logs.